src/111f0_len_860.c

```diff
@@ -37,7 +37,7 @@ void state_step_enter_world(void) {
                 gGameStatusPtr->isBattle = FALSE;
                 gGameStatusPtr->disableScripts = FALSE;
 
-                if (D_800A0948 == 0) {
+                if (gMapLoadFrom == LOAD_FROM_MAP) {
```

Comment:
I suspect D_800A0948 is actually a boolean "gLoadingDemoMap" (instead of sharing the values from this enum), considering the usage here and where its set.